cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -1812,7 +1812,9 @@ if(ARROW_WITH_PROTOBUF)
   else()
     set(ARROW_PROTOBUF_REQUIRED_VERSION "2.6.1")
   endif()
-  if(ARROW_ORC OR ARROW_WITH_OPENTELEMETRY)
+  if(ARROW_ORC
+     OR ARROW_WITH_OPENTELEMETRY
+     OR ARROW_SUBSTRAIT)
```

Could you keep them in alphabetical order?

```suggestion
     OR ARROW_SUBSTRAIT
     OR ARROW_WITH_OPENTELEMETRY)
```